src/gradcam.jl

```diff
@@ -0,0 +1,20 @@
+"""
+    GradCAM
+
+Analyze model by calculating the gradient of a neuron activation with respect to the input.
+This gradient is then used to calculate the Class Activation Map.
```

Comment:
The docstring should explain users how to use the method. Most users who haven't read the GradCAM paper will not be familiar with the terms "feature und adaptation layers", so the arguments should be explained. If necessary, we could add an example as well.

Reply:
I've added an initial overhaul of the doctoring with the commit 26ea2353a697baa4c2b0893cd8054aa4ae988b84.
For now I am not quite sure whether or not the descriptions suffice as they are now. I am especially unsure whether the `adaptation_layers` are sufficiently explained. I have similar concerns with the expression "compatible with a wide variety of CNN model-families" (taken from the paper). At the same time I don't want to have a too verbose doctstring.